libs/prebuilt/langgraph/prebuilt/memory_handler.py

```diff
@@ -0,0 +1,307 @@
+import json
+import math
+from typing import Callable, cast
+
+from langchain_core.language_models.chat_models import BaseChatModel
+from langchain_core.messages import AIMessage, BaseMessage, HumanMessage, SystemMessage
+from langchain_core.messages.utils import _get_message_openai_role, trim_messages
+from langchain_core.prompts.chat import ChatPromptTemplate, ChatPromptValue
+from langchain_core.runnables import Runnable, RunnableConfig
+
+from langgraph.store.base import BaseStore
+
+SUMMARIES_NS = ("summaries",)
+
+TokenCounter = Callable[[list[BaseMessage]], int]
+
+
+# TODO: does this need to be a Runnable?
+class BaseMemoryHandler(Runnable):
+    def invoke(  # type: ignore
```

Comment:
If we do `Runnables:

1) We'll want an async variant.
2) The interface for implementation should probably not be `invoke` but something like` _invoke` or `_process` so that `invoke` is reserved for instrumentation purposes


The reason to make this or not make this into a Runnable is if we want dedicated tracing for this in langsmith. But it's not clear to me that we do